### ISSUE TITLE
[Feature Request] Admonitions / Callouts

### ISSUE THREAD

**Issue by kiryph** [frankluebeck/GAPDoc#74]
Many documentation system have introduced special text blocks to draw the readers attention to critical information, called sometimes alerts, admonitions or callouts.

Examples:

- **GitHub Docs**, https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
- **AsciiDoc**, https://docs.asciidoctor.org/asciidoc/latest/blocks/admonitions/
- **MyST**, https://myst-parser.readthedocs.io/en/latest/syntax/admonitions.html
- **Obsidian Notes**, https://help.obsidian.md/Editing+and+formatting/Callouts
- **Doxygen Github Alerts**, https://www.doxygen.nl/manual/markdown.html#mddox_github_alerts
- **LaTeX pkg awesomebox**, https://ctan.org/pkg/awesomebox
...

Can something similar be added to GAPDoc?

**Comment by kiryph** [comment]
I have created a version allowing following syntax
```xml
<Section><Heading>Admonition Boxes</Heading>
<Label Name="ElevenBack"/>

[&see; <Ref Sect="Eleven"/>]<P/>

Five types of admonition boxes:<P/>

<Admonition Type="Note">
Useful information that users should know, even when skimming content.
</Admonition>

<Admonition Type="Tip">
Helpful advice for doing things better or more easily.
</Admonition>

<Admonition Type="Important">
Key information users need to know to achieve their goal.
</Admonition>

<Admonition Type="Warning">
Urgent info that needs immediate user attention to avoid problems.
</Admonition>

<Admonition Type="Caution">
Advises about risks or negative outcomes of certain actions.
</Admonition>

Custom captions:<P/>

<Admonition Type="Important">
<Caption>Important: Custom Caption</Caption>
Key information users need to know to achieve their goal.
</Admonition>

Other <Package>GAPDoc</Package> elements within admonition boxes are allowed:<P/>

<Admonition Type="Note">
<Caption>A note with <C>Log</C></Caption>
An example with syntax highlighting:
<Br/><Ignore> manually fixing space</Ignore>
<Log>
gap&gt; Sieve(100);
[ 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61,
  67, 71, 73, 79, 83, 89, 97 ]
gap&gt; Length(last);
25
</Log>
</Admonition>
```

 and looks like this:

## HTML

<img width="765" height="1583" alt="Image" src="https://github.com/user-attachments/assets/ca898f13-f418-4ecb-a059-57ca86eaff6c" />


## LaTeX (PDF)

<img width="920" height="1530" alt="Image" src="https://github.com/user-attachments/assets/8d5bc0b6-a272-4570-8418-185081a058ba" />


## Terminal

<img width="1167" height="657" alt="Image" src="https://github.com/user-attachments/assets/42510cbe-bdd4-42c7-a9c6-8809caf15691" />

This can be improved by drawing a box instead of using rulers.

**Comment by fingolfin** [comment]
@kiryph I like it -- care to make a pull request?

**Comment by frankluebeck** [comment]
I do not agree that this is a good idea. In my eyes this wastes space, makes the texts more difficult to read and is just unnecessary. Better think about improving the content of your manuals to make it easy for the readers to find what they want to know.
